addons/block_code/blocks/graphics/animationplayer_play.gd

```diff
@@ -0,0 +1,15 @@
+@tool
+extends BlockExtension
+
+const OptionData = preload("res://addons/block_code/code_generation/option_data.gd")
+
+
+func get_defaults_for_node(defaults: Dictionary, context_node: Node) -> Dictionary:
+	var animation_player = context_node as AnimationPlayer
+
+	if not animation_player:
+		return {}
```

Comment:
Shouldn't this return `defaults`?

Reply:
Thanks!

But I have a question while we're here: do you feel it's better if this function returns a complete defaults list (eg: returning `{}` means whatever defaults were there are now _gone_), or should it be returning just new defaults (so the caller is doing `defaults.merged(defaults, new_defaults)` itself). It doesn't make a difference either way in this case so I just went for the thing that makes the least assumptions.

Reply:
I feel like it should merge until we have evidence that it's better to override defaults from a base class.

Reply:
Also it turns out Dictionary.merged is new in Godot 4.3, so better to do the ugly alternative in one place anyway.

Reply:
OK, I see now that this returns additional defaults and BlockDefinition does the merging.